life_expectancy/tests/test_cleaning.py

```diff
@@ -1,17 +1,64 @@
-"""Tests for the cleaning module"""
+"""Tests for all modules"""
+
 import pandas as pd
+from ..load_save import load_data, save_data
+from ..cleaning import clean_data
+from ..main import main
+from unittest.mock import Mock, patch
+from . import OUTPUT_DIR, FIXTURES_DIR
+
+def _correct_data_types(df: pd.DataFrame) -> pd.DataFrame:
+    """Set correct data types.
+    
+    Args:
+        df (pd.DataFrame): The data.
+
+    Returns:
+        df (pd.DataFrame): The modified data.
+    """
+    for col in ['unit', 'sex', 'age', 'region']:
+        df[col] = df[col].astype('string')
+    df['year'] = df['year'].astype('int')
+    df['value'] = pd.to_numeric(
+        df['value'])
+    return df
+
+def test_clean_data(pt_life_expectancy_input: pd.DataFrame,
+                     pt_life_expectancy_cleaned_expected: pd.DataFrame):
+    """Test for the clean_data function
+    
+    Args:
+        pt_life_expectancy_input (pd.DataFrame): Fixture for function input.
+        pt_life_expectancy_cleaned_expected (pd.DataFrame): Fixture for function output.
+
+    """
+
+    pt_life_expectancy_cleaned_actual = clean_data(pt_life_expectancy_input, 'PT')
+    pd.testing.assert_frame_equal(pt_life_expectancy_cleaned_actual,
+                                  _correct_data_types(pt_life_expectancy_cleaned_expected))
 
-from life_expectancy.cleaning import main
-from . import OUTPUT_DIR
+@patch(".load_save.save_data.pandas.DataFrame.to_csv")
```

Comment:
```suggestion
@patch("pandas.DataFrame.to_csv")
```
Here, you patch pandas.DataFrame.to_csv directly since it's a method of the DataFrame class in the pandas module. You do not need to specify your module’s name in the patch.

Reply:
Hi Nuno, from what I read about patching, you're supposed to patch the method just in the module and function where it is tested, not the original method. I don't know why it does not work for me.